libgoal/libgoal.go

```diff
@@ -935,78 +935,81 @@ func MakeDryrunState(client Client, txnOrStxn interface{}, other []transactions.
 	return v2.DryrunRequestFromGenerated(&gdr)
 }
 
-// MakeDryrunStateGenerated function creates DryrunRequest data structure and serializes it into a file
+// MakeDryrunStateGenerated function creates generatedV2.DryrunRequest data structure and serializes it into a file
 func MakeDryrunStateGenerated(client Client, txnOrStxn interface{}, other []transactions.SignedTxn, proto string) (dr generatedV2.DryrunRequest, err error) {
 	var txns []transactions.SignedTxn
-	var tx *transactions.Transaction
-	if txn, ok := txnOrStxn.(transactions.Transaction); ok {
-		tx = &txn
+	if txnOrStxn == nil {
+		// empty input do nothing
+	} else if txn, ok := txnOrStxn.(transactions.Transaction); ok {
 		txns = append(txns, transactions.SignedTxn{Txn: txn})
 	} else if stxn, ok := txnOrStxn.(transactions.SignedTxn); ok {
-		tx = &stxn.Txn
 		txns = append(txns, stxn)
 	} else {
 		err = fmt.Errorf("unsupported txn type")
 		return
 	}
+
 	txns = append(txns, other...)
 	for i := range txns {
 		enc := protocol.EncodeJSON(&txns[i])
 		dr.Txns = append(dr.Txns, enc)
 	}
 
-	if tx.Type == protocol.ApplicationCallTx {
-		apps := []basics.AppIndex{tx.ApplicationID}
-		apps = append(apps, tx.ForeignApps...)
-		for _, appIdx := range apps {
-			var appParams generatedV2.ApplicationParams
-			if appIdx == 0 {
-				// if it is an app create txn then use params from the txn
-				appParams.ApprovalProgram = tx.ApprovalProgram
-				appParams.ClearStateProgram = tx.ClearStateProgram
-				appParams.GlobalStateSchema = &generatedV2.ApplicationStateSchema{
-					NumUint:      tx.GlobalStateSchema.NumUint,
-					NumByteSlice: tx.GlobalStateSchema.NumByteSlice,
-				}
-				appParams.LocalStateSchema = &generatedV2.ApplicationStateSchema{
-					NumUint:      tx.LocalStateSchema.NumUint,
-					NumByteSlice: tx.LocalStateSchema.NumByteSlice,
+	for _, txn := range txns {
+		tx := txn.Txn
+		if tx.Type == protocol.ApplicationCallTx {
```

Comment:
entire `if tx.Type == protocol.ApplicationCallTx {` block moved inside the loop without modifications